packages/shared/src/contexts/common.ts

```diff
@@ -1,2 +1,20 @@
+import React from 'react';
+import { Permissions } from 'webextension-polyfill';
+
 export const BOOT_LOCAL_KEY = 'boot:local';
 export const BOOT_QUERY_KEY = 'boot';
+
+export type RequestContentScripts = (data: {
+  origin: string;
+  skipRedirect?: boolean;
+}) => Promise<boolean>;
+
+export interface ExtensionContextData {
+  getContentScriptPermission?: () => Promise<boolean>;
+  requestContentScripts?: RequestContentScripts;
+  registerBrowserContentScripts?: () => Promise<void>;
+  getHostPermission?: () => Promise<boolean>;
+  requestHostPermissions?: Permissions.Static['request'];
+  origins?: string[];
+}
+export const ExtensionContext = React.createContext<ExtensionContextData>({});
```

Comment:
I'd love to here other opinions here if this is a practive we want to follow.
My concern would be it could be harder to see what actually gets used/not.

CC: @dailydotdev/web-team

Reply:
one potential argument for this is that we would no longer depend on the `QueryClient` provider being set first in order to use these -- which was the main reason I made the change, since we have a situation here where we need to make calls to these scripts before anything else.

Reply:
Yeah quite like the context overall, just the way in which some cases it might be emty/some it might be set might not be super clear.
Also perhaps we should consider having it as a proper context file in shared?
And then the one in extension will extend/override the data?

Reply:
fair point, it is actually always empty outside the extension (the provider isn't even mounted in the [webapp](https://github.com/dailydotdev/apps/blob/65b6c8cd36a5a45247682a1dc273caa620e54bf6/packages/webapp/pages/_app.tsx#L184-L202)

But I agree with your point that the way the `Context` and `Provider` are split here can be confusing -- I extracted it out into it's own `ExtensionContext` file in shared, as you suggested, which is then overridden in the extension